sos/report/plugins/ceph_osd.py

```diff
@@ -39,9 +39,62 @@ def setup(self):
 
         self.add_cmd_output([
             "ceph-disk list",
-            "ceph-volume lvm list",
+            "ceph-volume lvm list"
         ])
 
+        ceph_cmds = [
+           "bluestore bluefs available",
+           "config diff",
+           "config show",
+           "dump_blacklist",
+           "dump_blocked_ops",
+           "dump_historic_ops_by_duration",
+           "dump_historic_slow_ops",
+           "dump_mempools",
+           "dump_ops_in_flight",
+           "dump_op_pq_state",
+           "dump_osd_network",
+           "dump_reservations",
+           "dump_watchers",
+           "log dump",
+           "perf dump",
+           "perf histogram dump",
+           "objecter_requests",
+           "ops",
+           "status",
+           "version",
+        ]
+
+        osd_ids = []
+        out = self.exec_cmd('ps -u ceph -o args')
+
+        if out['status'] != 0:
+            return
+
+        # Get the running OSD process IDs from ps
+        osd_nums = [
+                m.split()[5] for m in out['output'].splitlines()
+                if 'ceph --id' in m
+        ]
+
+        for element in osd_nums:
+            osd_ids.append('osd.' + element)
+
+        containers_list = self.get_all_containers_by_regex("ceph-osd*")
+
+        if containers_list:
+            self.add_cmd_output([
+                self.fmt_container_cmd(
+                    containers_list[1][1], "ceph daemon %s %s"
```

Comment:
Can't `containers_list` contain exactly one element / one container? (if so, also `IndexError` could happen)

Reply:
agreed. this crept in through the changes made for https://github.com/sosreport/sos/pull/2736#discussion_r738476601 .. I guess 

self.fmt_container_cmd(
                    containers_list[0][1], "ceph daemon %s %s"

would work, since we just need any one container so the first one is as good as any.